### PR TITLE
Improve miral shell help text rendering (Fixes #799)

### LINKED ISSUE #799: Shells inconsistantly crash when output width is much larget than height
`miral-shell --x11-output 800x100` usually crashes. Less extreme aspect ratios crash less often, and by the time it gets to 2:1 they go away. There does not seem to be a problem with tall skinny outputs. `mir_demo_server` does the same thing.

### PR DESCRIPTION
**Pull request by wmww**
Improve miral shell help text rendering. (Fixes #799)

### CODE REVIEW

**Comment by wmww** [comment]
Fixes #799

**Comment by wmww** [comment]
See also #833

File: examples/example-server-lib/decoration_provider.cpp [hunk 1]
```diff
@@ -38,6 +38,8 @@
 #include <system_error>
 #include <iostream>
 
+namespace geom = mir::geometry;
```
Comment (AlanGriffiths) [inline examples/example-server-lib/decoration_provider.cpp]: My current preference is for `using namespace mir::geometry`, but that's non-blocking.

**Comment by wmww** [comment]
Thanks for the improvements!

**Pull request merged**